web/src/designer/state/migrateNotebook.ts

```diff
@@ -368,3 +372,17 @@ const updateVisibleTypes = (notebook: Notebook) => {
       Object.keys(notebook['ui-specification'].viewsets) || [];
   }
 };
+
+/**
+ * Ensure every field has a stable designerIdentifier
+ * (fall back to the field key if none existed).
+ */
+const updateDesignerIdentifiers = (notebook: Notebook) => {
```

Comment:
Can these identifiers be internal to the designer? Inject them when you load a notebook and filter them out when you export? I'd rather not add more to the external notebook schema if possible.